### PR TITLE
fix: refacto front state machine

### PR DESCRIPTION
**Pull request by clegirar**
Signed-off-by: clegirar <clemntgirard@gmail.com>

### CODE REVIEW

File: js/packages/redux/reducers/ui.reducer.ts [hunk 1]
```diff
@@ -218,54 +135,41 @@ const slice = createSlice({
...
-		},
-		setStateOpening: setStateOpeningFn,
-		setStateOpeningClients(state: UiState) {
-			state.appState = MESSENGER_APP_STATE.OPENING_WAITING_FOR_CLIENTS
+			Object.keys(initialState).map(k => {
+				if (['accounts'].indexOf(k) === -1) {
+					//@ts-ignore
+					state[k] = initialState[k as keyof UiState]
+				}
+			})
```
Comment (n0izn0iz) [inline js/packages/redux/reducers/ui.reducer.ts:156]:
```suggestion
			state.accounts = initialState.accounts
```

File: js/packages/screens/settings/DevTools/DevTools.tsx [hunk 2]
```diff
@@ -576,7 +573,7 @@ const BodyDevTools: React.FC<{}> = withInAppNotification(({ showNotification }:
 			<DropDownPicker
 				items={items}
 				defaultValue={i18next.language}
-				onChangeItem={(item: Item) => dispatch(setAccountLanguage(item.value))}
+				onChangeItem={async (item: Item) => await i18next.changeLanguage(item.value)}
```
Comment (n0izn0iz) [inline js/packages/screens/settings/DevTools/DevTools.tsx:576]: no need to await since a promise will be returned already

File: js/packages/store/messenger-effects/finishPreparingAccount.effect.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+export const finishPreparingAccount = async (
+	ui: UiState,
+	messenger: MessengerState,
+	conversations: Dictionary<beapi.messenger.IConversation>,
+	dispatch: AppDispatch,
+) => {
+	try {
+		await closeConvos(ui.messengerClient, conversations)
+
+		await i18next.changeLanguage(detectOSLanguage())
```
Comment (n0izn0iz) [inline js/packages/store/messenger-effects/finishPreparingAccount.effect.ts:70]: why change language here ?

File: js/packages/store/messenger-effects/initialLaunch.effect.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+			// Delete berty-backup account
+			const outFile =
+				RNFS.TemporaryDirectoryPath + `/berty-backup-${accountSelected.accountId}` + '.tar'
+			RNFS.unlink(outFile)
+				.then(() => {
+					console.log('File deleted')
+				})
+				.catch(() => {
+					console.log('File berty backup does not exist') // here
+				})
```
Comment (n0izn0iz) [inline js/packages/store/messenger-effects/initialLaunch.effect.ts:46]: why is this needed ?

**Review by n0izn0iz — CHANGES_REQUESTED** [review]
also maybe rename streamError to initError or something

File: js/packages/redux/reducers/ui.reducer.ts (hunk 1 above)
Comment (clegirar) [reply js/packages/redux/reducers/ui.reducer.ts:156]: nope it's for not reset accounts

File: js/packages/store/messenger-effects/finishPreparingAccount.effect.ts (hunk 3 above)
Comment (clegirar) [reply js/packages/store/messenger-effects/finishPreparingAccount.effect.ts:70]: yep no needed here, it was done before 👍

**Comment by clegirar** [comment]
streamError is not only set for an initialization error, you can have stream errors when you opening an account but also creating one, deleting one, importing one, updating one, so streamError seems fine like naming no ?

**Comment by n0izn0iz** [comment]
> streamError is not only set for an initialization error, you can have stream errors when you opening an account but also creating one, deleting one, importing one, updating one, so streamError seems fine like naming no ?

hmm, `globalError` maybe ?

**Comment by clegirar** [comment]
> hmm, globalError maybe ?
hmm, for you `streamError` it's only for messenger `EventsStream` ? That's what makes you ?

File: js/packages/redux/reducers/ui.reducer.ts (hunk 1 above)
Comment (n0izn0iz) [reply js/packages/redux/reducers/ui.reducer.ts:157]: sorry I read the condition as reverse, I get it now, maybe just do `if (k !== 'accounts')`

**Review by n0izn0iz — APPROVED** [review]
LGTM
